mobile/src/screens/survey/components/preferences-step/preferences-step.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+
+import {
+  Button,
+  Input,
+  ScrollView,
+  SurveyCategory,
+} from '#libs/components/components';
+import { useAppForm, useCallback, useFormController } from '#libs/hooks/hooks';
+import { type FormControllerRenderProps } from '#libs/types/types';
+import {
+  getSurveyCategories,
+  type SurveyInputDto,
+  surveyInputValidationSchema,
+  SurveyTextareaOptions,
+} from '#packages/survey/survey';
+import {
+  DEFAULT_SURVEY_PAYLOAD,
+  PREFERENCES_CATEGORIES,
+  TEXTAREA_MAX_LENGTH,
+  TEXTAREA_ROWS_COUNT,
+} from '#screens/survey/libs/constants';
+
+import { styles } from './styles';
+
+type Properties = {
+  onSubmit: (options: string[]) => void;
+};
+
+const PreferencesStep: React.FC<Properties> = ({ onSubmit }) => {
+  const {
+    watch,
+    control,
+    errors,
+    isValid,
+    handleSubmit: handleValidationSubmit,
+  } = useAppForm<SurveyInputDto>({
+    defaultValues: DEFAULT_SURVEY_PAYLOAD,
+    validationSchema: surveyInputValidationSchema,
+  });
+  const { field: optionsField } = useFormController({
+    name: 'options',
+    control,
+  });
+
+  const options: string[] = [
+    ...PREFERENCES_CATEGORIES,
+    SurveyTextareaOptions.OTHER,
+  ];
```

Comment:
Let's make a constant from SurveyTextareaOptions - PREFERENCES_OTHER_CATEGORY until we have more values additional options variables. Also just add it to PREFERENCES_CATEGORIES, we don't need options array
```suggestion
```